docs/onboarding_best_practices.rst

```diff
@@ -0,0 +1,165 @@
+######################
+Pinot has 2 modes of segment push. *APPEND and REFRESH*.
+
+- APPEND - every data push is an incremental one, and the data received is appended to the existing data.  
+- REFRESH - The existing data is completely replaced by the new data pushed. The new segment names need to be exactly the same for the existing segments to be replaced
+
+In general, if you push a segment with the same name, it will overwrite the previous segment (in either mode).
+
+4. Prefer long over string where possible
+##########################################
+Strings in Java are inefficient. We end up serializing/deserializing them during query execution a lot of time.
+
+Instead of storing a long string value, store an id, and use a stateless utility function for conversion. Similarly, for attributes of a dimension (*for example, companyId -> company details*) store only the id in Pinot, and use another key value store/pinot table for the lookup. This is applicable only if Pinot cannot handle desired latency/throughput, even after all other possible optimizations. The reason is that, inside Pinot, we don’t know the relation between different columns. If used in a query like group by (group by companyId, companyName, companySize)  we will try to gather all the combinations of these columns in order to get the group-by results. This is quite expensive inside Pinot, but with key-value store service (or even keep a separate Pinot table), this will be simply a value lookup, which is way cheaper, and can definitely give you better performance
+
+5. Sorted column
+###################
+Sort data on column if it will always be present in the queries. **Sorted index performs superior than inverted index.** 
+
+- For offline, Pinot cannot sort the data in segments that are pushed to pinot. This needs to be done prior to the segment creation, in the data preparation job. If the data is sorted before segment creation, the segment creation will detect and maintain this sorted column. 
+- For realtime, set the column in “sortedColumn” in table config, and it will be sorted in Pinot when data is converted to a Pinot segment.
+
+6. Multi value columns
+#######################
+Pinot supports multi value columns. Arrays in an avro will be treated as multi value columns, by setting `“singleValueField” : “false”` in the dimensionFieldSpec in Pinot schema. However, in general they are more costly for query execution, and ensure that they are being used only when needed.
+
+7. Inverted indexes
+###################
+For columns that will be used fairly frequently in your queries, enable inverted indexes. Inverted indexes are **more efficient and effective for high cardinality columns.** 
+
+8. Partitioning
+###################
+- For offline, if the data has been partitioned, Pinot can leverage this information for querying. This can be done by setting the partitioning function, number of partitions and column name in the segmentPartitionConfig in the table config 
```

Comment:
refer to the section where we describe partitioning pinot data (I hope there is one. Otherwise, start an empty section, we will fill it up)